design-proposals/dynamic-primary-pod-nic-name.md

```diff
@@ -538,3 +542,109 @@ The primary pod interface name is set by the container runtime (for example cri-
 
 - There is no K8s API to name the pod's primary NIC
 - There is no Multus API or config to name the pod's primary NIC
+
+# Appendix B - Support Primary Interface Name Change on Live Migration
+
+The original proposal treated the scenario where a migration target pod has a primary interface name different from the
+source as an edge case, and explicitly mentioned it as a non-goal.
+
+This limitation was raised as a concern, prompting a re-evaluation and the proposal of a mitigation strategy.
+
+## The Problem
+
+For several network bindings, the pod interface name is either directly or indirectly used to generate values that are set
+in the domain XML.
+
+Without intervention, a change to the primary interface name in the target pod will result in a migration failure due
+to a naming mismatch. Such a failure could, in turn, lead to an upgrade failure. 
+
+## Solution
+
+### Inferring the target pod's primary interface name
+
+All the alternatives listed bellow depend on knowing the target pod's primary interface name.
+However, the current logic in the VMI controller only provides the primary interface name of the source.
+
+To address this, an API change is proposed to enable the reporting if the target pod's primary interface name, allowing
+the necessary adjustments for migration to occur successfully: 
+
+```go
+type VirtualMachineInstanceMigrationState struct {
+...
+TargetPodPrimaryInterfaceName string `json:"targetPodPrimaryInterfaceName,omitempty"`
+...
+}
+```
+
+When the target pod is ready, the migration controller will:
+1. Identify its primary interface name (similar to the process used by the VMI controller).
+2. Report its value using the newly added field in the `VirtualMachineInstanceMigrationState` struct.
+
+### Adjusting the target primary interface name
+
+Since the tap associated with the VM's primary interface is always named `tap0` there is no need to adjust the domain XML
+when performing a live migration and the target pod interface name is changed.
+
+The target virt-handler will consider the `VMI.Status.MigrationState.TargetPodPrimaryInterfaceName` when executing
+the network setup flow.
+
+#### Pros
+- The addition of the migrationState field could be deferred until the need arises
+- The domain XML will not be modified in either source or target
+
+#### Cons
+- Does not handle network bindings which do not use tap
```

Comment:
We do not have any such core binding for the primary network.

Network binding plugins that intend to support such scenarios, will need to solve it themselves for now.
In the future, we do plan to support the mutation or observation of the domain xml on the migration target, giving a chance to touch it per the target needs. But this is futuristic.

Reply:
The con was removed.